egress/src/reports/windrose.rs

```diff
@@ -0,0 +1,1114 @@
+    if patches.is_empty() {
+        // Cannot query necessary data if either
+        // - there are no timeseries for wind speed or wind direction
+        // - no speed and direction patchwork timeseries overlap
+        return Ok(vec![]);
+    };
+
+    let conn = pool.get().await?;
+    let days = get_wind_days(patches, &params.months, &conn).await?;
+
+    Ok(days)
+}
+
+/// Query parameter for reports/windrose/{station_id} endpoint
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WindroseParams {
+    fromtime: DateTime<Utc>,
+    totime: DateTime<Utc>,
+    #[serde(default, deserialize_with = "optional_comma_separated")]
+    months: Option<Vec<i32>>,
+}
+
+/// Metadata returned with the response
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Metadata {
+    fromtime: DateTime<Utc>,
+    totime: DateTime<Utc>,
+    station_id: i32,
+    number_of_values: usize,
+    // TOOD: not sure this is what we want?
+    #[serde(skip_serializing_if = "Option::is_none")]
+    months: Option<Vec<i32>>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Axis {
+    /// Axis labels
+    #[serde(skip_deserializing)]
+    labels: &'static [&'static str],
+    /// 1D histogram values
+    pub sums: Vec<f64>,
+}
+
+/// Response from reports/windrose/{station_id} endpoint
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WindroseResp {
+    pub wind_speed: Axis,
+    pub wind_direction: Axis,
+    pub extras: WindCategories,
+    pub table: Vec<Vec<f64>>,
+    pub metadata: Metadata,
+}
+
+pub async fn windrose_handler(
+    Path(station_id): Path<i32>,
+    Query(params): Query<WindroseParams>,
+    State(pools): State<DbPools>,
+    State(tables): State<PatchworkTables>,
+    Extension(roles): Extension<Option<Vec<i32>>>,
+) -> Result<Json<WindroseResp>, (StatusCode, String)> {
+    let roles = roles.unwrap_or_default();
+
+    // NOTE: given how permits work at the moment, open and restricted are mutually exclusive
+    let (open_data, restricted_data) = tokio::try_join!(
+        fetch_wind_data(station_id, &params, &roles, pools.open, tables.open),
+        fetch_wind_data(
+            station_id,
+            &params,
+            &roles,
+            pools.restricted,
+            tables.restricted
+        ),
+    )
+    .map_err(internal_error)?;
+
+    let days = match (open_data.is_empty(), restricted_data.is_empty()) {
+        (false, _) => open_data,
+        (_, false) => restricted_data,
+        (true, true) => {
+            return Err((
+                StatusCode::NOT_FOUND,
+                "no data found for this station".to_string(),
+            ))
+        }
+    };
+
+    // TODO: spawn sync thread here?
+    let windrose = Windrose::new_from_days(SPEED_AXIS, DIRECTION_AXIS, days);
+
+    let metadata = Metadata {
+        station_id,
+        fromtime: params.fromtime,
+        totime: params.totime,
```

Comment:
Seems deceptive to me that we don't indicate the timeranges that are actually covered by data